src/Sylius/Bundle/ApiBundle/Serializer/TranslatableLocaleKeyDenormalizer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Component\Resource\Model\TranslatableInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+
+/** @experimental */
+final class TranslatableLocaleKeyDenormalizer implements ContextAwareDenormalizerInterface, DenormalizerAwareInterface
+{
+    use DenormalizerAwareTrait;
+
+    private const ALREADY_CALLED = 'sylius_translatable_locale_key_denormalizer_already_called';
```

Comment:
Will this denormalizer be called on relations, if one translatable has another, like product and main taxon?

Is it necessary for this denormalizer to be executed multiple times?

Reply:
#15535